### PR TITLE
Accept supports declarations for express payment style controls + merchant ux improvements in the editor

### PR DESCRIPTION
**Pull request by alexflorisca**
### Changes proposed in this Pull Request:

This PR is part of [[Feature] Express Checkout Improvements](https://github.com/woocommerce/woocommerce/pull/50791)

In #50791, some new styling controls were added for the express payment buttons. There have been various discussions across slack and PRs about the best way to handle the fact that this also requires changes in the various payment extensions. Meanwhile, there will be a situation where some express payment buttons don't support the controls.

This PR aims to improve the merchant UX in the editor slightly in these situations by:
- Adding the ability for payment methods to declare support for the various styling controls
- Adding a "Beta" label to the "Apply uniform styles" heading
- Hiding the controls if no express payment methods support them
- Displaying a warning to merchants informing them that not all payment methods may support the style controls.

Screenshot 2024-09-11 at 15 47 42

### How to test the changes in this Pull Request:

**Prerequisites**
1. Install & Activate Woocommerce Payments from this zip file [woocommerce-payments.zip](https://github.com/user-attachments/files/16965496/woocommerce-payments.zip)
2. Install the [WooCommerce Payments Dev Tools plugin](https://github.com/Automattic/woocommerce-payments-dev-tools). 
3. Check the `Force the platform_checkout_eligible flag (aka WooPay eligible) in the account cache to be` setting and set it to `true`
4. Check the `Enable the WooPay Express Checkout button` box
5. Enable the Woo Payments payment gateway and make sure the the WooPay
6. Install WooCommerce Stripe Payment Gateway from this zip [woocommerce-gateway-stripe.zip](https://github.com/user-attachments/files/16965515/woocommerce-gateway-stripe.zip)
7. Activate the Stripe payment gateway
8. Install [PayPal Payments Plugin](https://woocommerce.com/products/woocommerce-paypal-payments/)
9. Activate The PayPal payment gateway.
10. Make sure Standard Payments is enabled and that the smart buttons are shown on the Cart & Express Checkout  locations.

**Testing**
1. Go to the Checkout page in the editor
2. Click on the express checkout block. Notice the controls in the right sidebar match the screenshot above. Makre sure the beta label and the warning are present
3. In a different tab, head to WooCommerce Payment settings (/wp-admin/admin.php?page=wc-settings&tab=checkout) & disable WooPayments
4. Refresh the checkout page in the editor and click on the express checkout block. You should **not** see the border radius control anymore.
5. In your WooCommerce Settings tab, disable Stripe
6. Refresh the checkout page in the editor and click on the express checkout block. The entire "Button Settings" panel should be hidden.
7. Smoke test the checkout on the front-end and complete an order with an express payment method. Make sure no editor stylesheets are loaded on the front-end.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

UX improvements to the express payment block in the editor
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/block-settings/index.tsx [hunk 1]
```diff
@@ -44,130 +35,4 @@ export const BlockSettings = ( {
 	);
 };
 
-const ExpressPaymentButtonStyleControls = ( {
```
Comment (alexflorisca) [inline plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/block-settings/index.tsx:47]: All the express payment controls were moved to the file above `express-payment-settings.tsx`

**Comment by senadir** [comment]
Not sure if related to this one or already fixed, but the sidebar is slightly broken by very long plugin names

image
image

**Review by senadir — APPROVED** [review]
This seems to match the testing steps, but I do wish it was a bit smarter in detecting incompatibility.

When I had WooPay disabled, the border radius still showed yet no button supported it.

Also with the switch to SVG, button sizes are never correct and always looking weird

Stripe:
image

**Comment by alexflorisca** [comment]
> Not sure if related to this one or already fixed, but the sidebar is slightly broken by very long plugin names

Not related but I know about it and will fix on the feature branch

> When I had WooPay disabled, the border radius still showed yet no button supported it.

As in, WooPay was disabled, but Google/Apple pay wasn't? If so, this is because I set the google and apple pay buttons to support the border radius in the WooPayments zip that I added for testing. Which they do, but in reality, extensions will add the supports capability only once they've rendered buttons instead of images in the editor. So this should be taken care of.

> but I do wish it was a bit smarter in detecting incompatibility.

Do you have any other suggestions?

> Also with the switch to SVG, button sizes are never correct and always looking weird

I will fix this on the feature branch as well. This is some code I added to better style images, but I think since we removed settings the controls on the images, this doesn't work so well. Thanks for pointing this out.

**Comment by senadir** [comment]
> As in, WooPay was disabled, but Google/Apple pay wasn't? If so, this is because I set the google and apple pay buttons to support the border radius in the WooPayments zip that I added for testing. Which they do, but in reality, extensions will add the supports capability only once they've rendered buttons instead of images in the editor. So this should be taken care of.

That's I think is enough, as long as it's at button level, it should be fine.

**Pull request merged**